napari/_qt/widgets/_tests/test_qt_extension2reader.py

```diff
@@ -122,13 +119,16 @@ def test_directory_readers_not_in_dropdown(
     assert 'Directory Reader' not in all_dropdown_items
 
 
+@pytest.mark.xfail(
+    reason="This is predicated on napari only having npe1 readers"
```

Comment:
This one at least we can fix - I *think* the full test should be:

```python
def test_filtering_readers(
    extension2reader_widget, qtbot, tmp_reader, mock_npe2_pm
):
    tmp_reader(mock_npe2_pm, 'npy-reader', filename_patterns=['*.npy'])
    tmp_reader(mock_npe2_pm, 'tif-reader', filename_patterns=['*.tif'])

    widget = extension2reader_widget()

    assert widget._new_reader_dropdown.count() == 2
    widget._filter_compatible_readers('*.npy')
    assert widget._new_reader_dropdown.count() == 1
    all_dropdown_items = [
        widget._new_reader_dropdown.itemText(i)
        for i in range(widget._new_reader_dropdown.count())
    ]
    assert all_dropdown_items == ['npy-reader']
```

Reply:
no biggie.  I'm gonna skip it here.  i refactored the fixtures in https://github.com/napari/napari/pull/4669